Storage/tests/Unit/data/retry_tests.json

```diff
@@ -0,0 +1,281 @@
+{
+    "retryTests": [
+      {
+        "id": 1,
```

Comment:
Hypothetical Q, how do we ensure that conformance tests JSON is upto date across libraries? Maybe adding a `version` could work? Seems others also don't have it, [ref](https://github.com/googleapis/google-cloud-ruby/blob/main/google-cloud-storage/conformance/v1/retry_tests.json).

Reply:
I think this should have been the case.
If storage had an auto generated layer, this would have been easy.

But for now this end result is fine. After all if there are other scenarios that need to be added in conformance testing it is gonna be a bigger task, and might not be as simple as just editing the JSON.